plugins/inputs/intel_rdt/intel_rdt.go

```diff
@@ -334,12 +346,24 @@ func (r *IntelRDT) processOutput(cmdReader io.ReadCloser, processesPIDsAssociati
 
 func shutDownPqos(pqos *exec.Cmd) error {
 	if pqos.Process != nil {
-		err := pqos.Process.Signal(os.Interrupt)
-		if err != nil {
-			err = pqos.Process.Kill()
+		if pqos.Path == "/bin/sh" { // the process is running under sudo
+			cmd := exec.Command("sudo", "killall", "-SIGINT", "pqos")
```

Comment:
I don't like that to be honest. Imagine you have multiple processes running on the host using the `pqos` command. Now you also potentially kill the `pqos` of other processes. That's not very nice. :-) Why can't you just kill the command you started? You can get the PID from `pqos.Process.Pid`.

Reply:
Challenge with that one is that the Pid is for the parent process, 'sh', and what we want is to send a -SIGINT to the pqos child process.  As far as I'm aware, we can't actually run pqos more than once anyway, so it's in theory not a big issue, but a fair point regardless in case that changes.

Reply:
Can you please again outline on why we need that `sh` thing in the first place? If we really need it, I fear we have to use `gopsutils` to determine the children of the `sh` process via [Process.Children](https://pkg.go.dev/github.com/shirou/gopsutil@v3.21.6+incompatible/process#Process.Children) and also send the signal there.